### DISCUSSION TITLE
Best way to update PGMA-Modernized Agent to most recent version?

### DISCUSSION THREAD

**Discussion by carloschristian** [CodyBerenson/PGMA-Modernized#217] [Q&A]
Hello! I know this is probably, the most basic question with the most obvious answer, but can anyone detail the best ways in which to update the PGMA agent? Is the only way to update everything to the newest version to download everything again and replace what is already loaded into your Plex plugin directory? 

Thank you! Your help is much appreciated :)

**Comment by JPH71** [comment]
Short answer is yes...
A lu of. Hanges have been made to the basic infrastructure..And the
preference settings have changed a lot.

Download all the bundles...
Load up each agents preferences then save
Restart plex media server and you are good to go

**Reply by carloschristian** [reply]
Hi, @JPH71 - 

Thank you so much for your speedy response! How do I load up each agent's preferences and then save them? I see some XML files in /Volumes/PlexMediaServer/AppData/Plex Media Server/Plug-in Support/Preferences that look like my current agent preferences. Since I'm not overwriting those XML files, I assume I should be fine, but I want to be sure. 

Thanks again! I can't thank you enough for your help <3

**Comment by JPH71** [comment]
Oops, sorry for not replying earlier...
Click on the spanner at the top right of the player
In the left column sxroll all the way done to Agents
On the right side click on  GayAdult
You will see all the agents listed
For each agent click on its gear wheel
Se your preferences and SA e.
When finished restart plex media server...

**Comment by JPH71** [comment]
Did it work??

**Reply by carloschristian** [reply]
Ahhh I'm a total dick for not responding to you - apologies, but I can confirm your detailed steps did indeed work :) 

Thank you !

**Comment by JPH71** [comment]
No worries man...
Remember when you download the latest bundles that Cody puts on, you will
have to update the preferences again as there have been some new changes....

Jason x
